### PR TITLE
Refactor part 2: Move array params to separate classes, closes #118

### LINKED ISSUE #118: Refactor part 2: Move array params to separate classes
Move array params to separate classes:
- defaultParams
- ParsedOptions

### PR DESCRIPTION
**Pull request by sadok-f**

### CODE REVIEW

File: app.php [hunk 1]
```diff
@@ -8,10 +8,8 @@
 
 $app = new Silex\Application();
 
-/**
- * Application parameters
- */
-$app['params'] = yaml_parse(file_get_contents(__DIR__.'/config/parameters.yml'));
+/** @var \Core\Entity\AppParameters $app['params'] */
+$app['params'] = new \Core\Entity\AppParameters(__DIR__.'/config/parameters.yml');
```
Comment (baamenabar) [inline app.php]: is it OK to pas a filepath to initialise a Class??

File: app.php [hunk 2]
```diff
@@ -25,7 +23,7 @@
 };
 
 ErrorHandler::register();
-$exceptionHandler = ExceptionHandler::register($app['params']['debug']);
+$exceptionHandler = ExceptionHandler::register($app['params']->get('debug'));
```
Comment (baamenabar) [inline app.php]: $app['params'] 
is an object now??

File: app.php (hunk 1 above)
Comment (sadok-f) [reply app.php]: Yes :)

File: app.php (hunk 2 above)
Comment (sadok-f) [reply app.php]: yes it is :)
better right ?

**Pull request merged**